pkg/cmd/pipeline/start.go

```diff
@@ -222,6 +276,408 @@ func (opt *startOptions) run(pipeline *v1beta1.Pipeline) error {
 	return opt.startPipeline(pipeline)
 }
 
+func (opt *startOptions) runWithResolver(args []string) error {
+	cs, err := opt.cliparams.Clients()
+	if err != nil {
+		return err
+	}
+
+	var pipelineName string
+	if len(args) > 0 {
+		pipelineName = args[0]
+	}
+
+	// Special case: if resolvertype is "remote", find and rerun the latest PipelineRun with any resolver
+	if opt.ResolverType == "remote" {
+		return opt.runWithRemoteResolver(cs, pipelineName)
+	}
+
+	if opt.Last {
+		// Case: --resolvertype and --last
+		return opt.runWithResolverAndLast(cs, pipelineName)
+	}
+	// Case: --resolvertype only
+	if pipelineName == "" {
+		return errors.New("pipeline name is required when using --resolvertype flag")
+	}
+	return opt.runWithResolverOnly(cs, pipelineName)
+}
+
+func (opt *startOptions) runWithResolverOnly(_ *cli.Clients, pipelineName string) error {
```

Comment:
all those methods have a lot of duplications.. any way  to streamline this ? by extracting it to a helper method like createObjectMeta(lastPipelineRun, prefixName)